File: strategy_pattern/ruby/rocket_duck.rb
# frozen_string_literal: true

require_relative 'duck'
# A model rocket shaped like a duck
class com.randylayne.designpatterns.strategy.RocketDuck < com.randylayne.designpatterns.strategy.Duck
  def initialize
    super(com.randylayne.designpatterns.strategy.QuackBehavior::com.randylayne.designpatterns.strategy.MuteQuack, com.randylayne.designpatterns.strategy.FlyBehavior::RocketPowered)
  end

  def display
    puts "I'm a rocket powered duck"
  end
end

File: strategy_pattern/java/com/randylayne/designpatterns/strategy/RocketDuck.java
package com.randylayne.designpatterns.strategy;

public class RocketDuck extends Duck {
   public RocketDuck() {
      flyBehavior = new FlyRocketPowered();
      quackBehavior = new MuteQuack();
   }

   @Override
   public void display() {
      System.out.println("I'm a rocket powered duck");
   }
}
